src/ray/core_worker/lib/java/org_ray_runtime_RayNativeRuntime.cc

```diff
@@ -39,9 +39,23 @@ JNIEXPORT jlong JNICALL Java_org_ray_runtime_RayNativeRuntime_nativeInitCoreWork
          const std::vector<std::shared_ptr<ray::RayObject>> &args,
          const std::vector<ObjectID> &arg_reference_ids,
          const std::vector<ObjectID> &return_ids,
-         std::vector<std::shared_ptr<ray::RayObject>> *results) {
+         std::vector<std::shared_ptr<ray::RayObject>> *results,
+         const ray::WorkerID &worker_id) {
         JNIEnv *env = local_env;
+        if (!env) {
+          // Attach the native thread to JVM.
+          auto status =
+              jvm->AttachCurrentThreadAsDaemon(reinterpret_cast<void **>(&env), NULL);
+          RAY_CHECK(status == JNI_OK) << "Failed to get JNIEnv. Return code: " << status;
+          local_env = env;
```

Comment:
why is this required?

Reply:
`env` is thread local now. 
It's null in the concurrent thread pool.